File: src/natives/python/shared_memory_request.py
import request_generator


class SharedMemoryRequest(request_generator.RequestGenerator):
    def __init__(self, client):
        super(SharedMemoryRequest, self).__init__(client)
        self.type = 'shared_memory'
        self.device = 'shared_memory'

    def get(self, namespace, variable_name):
        self.action = 'get'
        self.params = [namespace, variable_name]
        return self.send_request()

    def set(self, namespace, variable_name, value):
        self.action = 'set'
        self.params = [namespace, variable_name, str(value)]
        return self.send_request()

    def delete(self, namespace, variable_name):
        self.action = 'del'
        self.params = [namespace, variable_name]
        return self.send_request()

    def get_instance(self, namespace):
        return SharedMemoryInstance(self, namespace)

class SharedMemoryInstance(object):
    """
        A small class to conveniently call SharedMemoryRequest class with fixed parameters.

    """
    def __init__(self, shared_memory_request, namespace):
        super(SharedMemoryInstance, self).__init__()
        self.shared_memory_request = shared_memory_request
        self.namespace = namespace

    def get(self, variable_name):
        return self.shared_memory_request.get(self.namespace, variable_name)

    def set(self, variable_name, value):
        return self.shared_memory_request.set(self.namespace, variable_name, value)

    def delete(self, variable_name):
        return self.shared_memory_request.delete(self.namespace, variable_name)

File: src/natives/csharp/source/IPC/SharedMemoryRequest.cs
using Newtonsoft.Json.Linq;
using Repeat.ipc;
using Repeat.userDefinedAction;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Repeat.IPC {
    public class SharedMemoryRequest : RequestGenerator {

        public SharedMemoryRequest(RepeatClient client) : base(client) {
            this.Type = "shared_memory";
            this.Device = "shared_memory";
        }

        public string GetVar(string nameSpace, string name) {
            Action = "get";
            ParamInt.Clear();
            ParamStrings.Clear();
            ParamStrings.Add(nameSpace);
            ParamStrings.Add(name);

            JToken result = SendRequest();
            if (result == null) {
                return null;
            }

            return result.Value<string>();
        }

        public string SetVar(string nameSpace, string name, string value) {
            Action = "set";
            ParamInt.Clear();
            ParamStrings.Clear();
            ParamStrings.Add(nameSpace);
            ParamStrings.Add(name);
            ParamStrings.Add(value);

            JToken result = SendRequest();
            if (result == null) {
                return null;
            }

            return result.Value<string>();
        }

        public string DelVar(string nameSpace, string name) {
            Action = "del";
            ParamInt.Clear();
            ParamStrings.Clear();
            ParamStrings.Add(nameSpace);
            ParamStrings.Add(name);

            JToken result = SendRequest();
            if (result == null) {
                return null;
            }

            return result.Value<string>();
        }

        public SharedMemoryInstance GetInstance(string nameSpace) {
            return new SharedMemoryInstance(this, nameSpace);
        }
    }
}
